UI/Theme.cpp

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) 2013- PPSSPP Project.
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 2.0 or later versions.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License 2.0 for more details.
+
+// A copy of the GPL 2.0 should have been included with the program.
+// If not, see http://www.gnu.org/licenses/
+
+// Official git repository and contact information can be found at
+// https://github.com/hrydgard/ppsspp and http://www.ppsspp.org/.
+
+#include "ppsspp_config.h"
+
+#include <algorithm>
+
+#include "UI/Theme.h"
+
+#include "Common/File/FileUtil.h"
+#include "Common/File/VFS/VFS.h"
+#include "Common/Data/Format/IniFile.h"
+
+#include "Core/Config.h"
+
+#include "Common/UI/View.h"
+#include "Common/UI/Context.h"
+#include "Core/System.h"
+
+struct themeInfo {
+	std::string name;
```

Comment:
I feel like it'd be really nice (maybe not in this pull) to also allow overriding the main zim (if not the font) per theme.  Since we read all the metadata from there, this would allow customizing icons, etc.

It might also be nice to have a default ini in the repo or somewhere that has the default theme (but also like you have, still work without it.)

-[Unknown]

Reply:
ZIM override would allow for custom touch controls too that got requested quite a few times.

Mhhhh, I got the feeling this the way to go rather than hue shift (kinda fancier with the slider but that's it).

Reply:
Now I kinda feel .. why not both? You pick a theme, and then you can shift it if you don't like its colors.. Maybe it's overkill.

Reply:
Such use cases could be covered by a web app to create or edit themes. We could also change the lightness value without changing the image redering as it happens on the vertex stage.
We could get the web interface make the ZIM file in the future too or thing like this to make it easier for the user.

But we could have both in the end, never disliked overkills in software to begin with :P

Reply:
These shaders are at least quite a bit less complicated than some of the stuff we generate sometimes, so I think they'll be fine. I will test them on Mali-400 though :)